experimental/auxiliary_sampling/datasets.py

```diff
@@ -19,142 +19,27 @@
-              'Tertiary Structure. The data set is taken from CASP 5-9. There '
-              'are 45730 decoys and size varying from 0 to 21 armstrong.'),
-        label='RMSD',
-        excluded=[]),
-    'wine': DataSpec(
-        path=os.path.join(UCI_BASE_DIR, 'wine.csv'),
-        desc=('The dataset is related to red variant of the Portuguese '
-              '"Vinho Verde" wine. **NB contains red wine examples only**'),
-        label='quality',
-        excluded=[]),
-    'yacht_hydrodynamics': DataSpec(
-        path=os.path.join(UCI_BASE_DIR, 'yacht_hydrodynamics.csv'),
-        desc=('Delft data set, used to predict the hydodynamic performance of '
-              'sailing yachts from dimensions and velocity.'),
-        label='Residuary resistance per unit weight of displacement',
-        excluded=[])
-}
-
-
-def get_uci_data(name):
-  """Returns an array of features and a vector of labels for dataset `name`."""
-  spec = DATA_SPECS.get(name)
-  if spec is None:
-    raise ValueError('Unknown dataset: {}. Available datasets:\n{}'.format(
-        name, '\n'.join(DATA_SPECS.keys())))
-  with tf.io.gfile.GFile(spec.path) as f:
-    df = pd.read_csv(f)
-  labels = df.pop(spec.label).as_matrix().astype(np.float32)
-  for ex in spec.excluded:
-    _ = df.pop(ex)
-  features = df.as_matrix().astype(np.float32)
-  return features, labels
-
-
 def load(name, session):
-  """Load dataset (UCI or mnist, fashion_mnist, cifar10) as numpy array."""
-
-  if name in ['mnist', 'fashion_mnist', 'cifar10']:
-    train_ds = tfds.load(
-        name, split=tfds.Split.TRAIN, batch_size=-1, as_supervised=True)
-    x_train, y_train = session.run(train_ds)
-    n_train = x_train.shape[0]
-    test_ds = tfds.load(
-        name, split=tfds.Split.TEST, batch_size=-1, as_supervised=True)
-    x_test, y_test = session.run(test_ds)
-    # Cifar10 is standardized. The other datasets are scaled to the [0, 1]
-    # interval.
-    if name == 'cifar10':
-      x_mean = np.mean(x_train, axis=(0, 1, 2))
-      x_std = np.std(x_train, axis=(0, 1, 2))
-      x_train = (x_train - x_mean) / (x_std + 1e-10)
-      x_test = (x_test - x_mean) / (x_std + 1e-10)
-    else:
-      x_min, x_max = np.amin(x_train), np.amax(x_train)
-      x_train = (x_train - x_min) / (x_max + 1e-10)
-      x_test = (x_test - x_min) / (x_max + 1e-10)
-  else:
-    x, y = get_uci_data(name)
-    if len(y.shape) == 1:
-      y = y[:, None]
-    train_test_split = 0.8
-    random_permutation = np.random.permutation(x.shape[0])
-    n_train = int(x.shape[0] * train_test_split)
-    train_ind = random_permutation[:n_train]
-    test_ind = random_permutation[n_train:]
-    x_train, y_train = x[train_ind, :], y[train_ind, :]
-    x_test, y_test = x[test_ind, :], y[test_ind, :]
-
-    # Standardize
-    x_mean, x_std = np.mean(x_train, axis=0), np.std(x_train, axis=0)
-    y_mean = np.mean(y_train, axis=0)
+  """Load dataset (mnist, fashion_mnist, cifar10) as numpy array."""
```

Comment:
We want to use the same function to fetch the datasets across all methods, including baselines, to make sure that the splits are consistent. I am guessing this code is for the image datasets and we want to use the data loader in deterministic/ for the UCI datasets.

Reply:
Yep, that's correct. In the uncertainty baselines codebase, there's common boilerplate code for all methods, including using the same utility function for loading datasets. We do try to keep methods distinct however, so there's a fine balance between code duplication and too many shared functions/"high-level" code.

Reply:
Yep, that's correct. In the uncertainty baselines codebase, there's common boilerplate code for all methods, including using the same utility function for loading datasets. We do try to keep methods distinct however, so there's a fine balance between code duplication and too many shared functions/"high-level" code.